### ISSUE TITLE
Compatibility Issue with Firebase and GoogleUtilities Versions

### ISSUE THREAD

**Issue by Yusufihsangorgel** [juliansteenbakker/mobile_scanner#1173]
I’m encountering dependency conflicts in my Flutter project when using the mobile_scanner package alongside firebase_core and other Google Utilities. The specific conflict arises with the versions of GoogleUtilities required by mobile_scanner conflicting with those required by firebase_core.

Environment:
- Flutter Version: [specify your Flutter version]
- mobile_scanner Version: 5.2.2
- firebase_core Version: 11.0.0 (firebase_core: ^3.4.0 in pubspec.yaml)
- iOS Deployment Target: iOS 15.0 (specified in Podfile)

Issue Detail:
mobile_scanner depends on GoogleUtilities/UserDefaults with a version constraint that allows versions < 8.0, >= 7.13.0, which is incompatible with the version required by firebase_core (GoogleUtilities/Logger ~> 8.0). This incompatibility prevents a successful pod installation and causes version conflicts during the dependency resolution process.

Error Message:
When attempting to install pods, I receive the following error message:
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities/Logger":
  In snapshot (Podfile.lock):
    GoogleUtilities/Logger (= 7.13.3)

  In Podfile:
    firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 3.4.0, which depends on
      Firebase/CoreOnly (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseCore (= 11.0.0) which depends on
          GoogleUtilities/Logger (~> 8.0)

Request:
Could the dependencies for GoogleUtilities within mobile_scanner be updated or managed to support newer versions, aligning with other common Firebase and Flutter packages? This would greatly assist in resolving conflicts and ensure smoother integration for projects using a broader range of dependencies.

**Comment by anbocas** [comment]
Same issue with firebase_messaging.

```
firebase_messaging (from .symlinks/plugins/firebase_messaging/ios) was resolved to 15.1.0, which depends on
      Firebase/Messaging (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseMessaging (~> 11.0.0) was resolved to 11.0.0, which depends on
          GoogleDataTransport (~> 10.0)

    mobile_scanner (from .symlinks/plugins/mobile_scanner/ios) was resolved to 5.2.2, which depends on
      GoogleMLKit/BarcodeScanning (~> 6.0.0) was resolved to 6.0.0, which depends on
        GoogleMLKit/MLKitCore (= 6.0.0) was resolved to 6.0.0, which depends on
          MLKitCommon (~> 11.0.0) was resolved to 11.0.0, which depends on
            GoogleDataTransport (< 10.0, >= 9.4.1)
```

**Comment by navaronbracke** [comment]
Duplicate of https://github.com/juliansteenbakker/mobile_scanner/issues/1165
